ndc-reference/bin/reference/main.rs

```diff
@@ -144,13 +149,48 @@ async fn main() {
         ))
         .with_state(app_state);
 
-    // run it with hyper on localhost:8100
-    axum::Server::bind(&"0.0.0.0:8100".parse().unwrap())
-        .serve(app.into_make_service())
-        .await
-        .unwrap();
+    // Start the server on `localhost:<PORT>`.
+    // This says it's binding to an IPv6 address, but will actually listen to
+    // any IPv4 or IPv6 address.
+    let host = net::IpAddr::V6(net::Ipv6Addr::UNSPECIFIED);
+    let port = env::var("PORT")
```

Comment:
I changed this to `PORT` to avoid any confusion that we might be implementing the deployment spec, and factored out a `shutdown_handler()` function, since `main()` gets shown in full in the book. Otherwise, looks good to me, merging now, thanks!